src/components/publishing/article.tsx

```diff
@@ -52,6 +52,18 @@ class Article extends React.Component<ArticleProps, ArticleState> {
     }
   }
 
+  @track((props, [e]) => ({
+    action: "Article Impression",
+    article_id: props.article.id,
+    destination_path: window.location.pathname,
+    // TODO: What do we put here? According to analytics there are the valid types
+    // [newsletter signup, toc, artist follow, image set, article callout, social, related article]
+    impression_type: "related_article",
```

Comment:
I'm not sure what to do about impression type here? This fires when the page loads essentially.

Reply:
Ah so these should fire each time an interactive component comes into the viewport, but only once per component. By interactive component, I mean anything with a link, signup, or button (i.e video, image set, view fullscreen, etc). We use this in combination with click tracking to determine engagement on parts of an article. 

The tricky thing is knowing when the component is in view with React. We do it this crazy way with jQuery currently: https://github.com/artsy/force/blob/master/desktop/analytics/article_impressions.js

Either way it seems like there needs to be clarification of _which_ components we need impression/click tracking on. Feel free to remove this part for now and I can refine that list later with owen or will.

Reply:
Ahhh, that makes more sense now 😛  I'll remove this, and no worries—I'll sync up with Will/Owen on clarification.